pkg/cmd/roachtest/run.go

```diff
@@ -327,3 +327,101 @@ func testRunnerLogger(
 	shout(ctx, l, os.Stdout, "test runner logs in: %s", runnerLogPath)
 	return l, teeOpt
 }
+
+// runOperations sequentially runs operations matched by the passed-in filter.
+func runOperations(register func(registry.Registry), filter *registry.TestFilter) error {
+	//lint:ignore SA1019 deprecated
+	rand.Seed(roachtestflags.GlobalSeed)
+	r := makeTestRegistry()
+
+	register(&r)
+	cr := newClusterRegistry()
+	stopper := stop.NewStopper()
+	defer stopper.Stop(context.Background())
+	runner := newTestRunner(cr, stopper)
+	runner.config.skipClusterStopOnAttach = true
+	runner.config.skipClusterWipeOnAttach = true
+
+	clusterType := roachprodCluster
+	bindTo := ""
+
+	if runtime.GOOS == "darwin" {
+		// This will suppress the annoying "Allow incoming network connections" popup from
+		// OSX when running a roachtest
+		bindTo = "localhost"
+	}
+
+	opt := clustersOpt{
+		typ:         clusterType,
+		clusterName: roachtestflags.ClusterNames,
+		// Precedence for resolving the user: cli arg, env.ROACHPROD_USER, current user.
+		user:      getUser(roachtestflags.Username),
+		cpuQuota:  roachtestflags.CPUQuota,
+		clusterID: roachtestflags.ClusterID,
+	}
+	opt.debugMode = DebugKeepAlways
+
+	if err := runner.runHTTPServer(roachtestflags.HTTPPort, os.Stdout, bindTo); err != nil {
```

Comment:
I don't think we need this?

Reply:
We do want an HTTP endpoint; it can live in `drt-run` as opposed to here, but I'll remove it here to speed this PR along.